core/node/eth_sender/src/eth_tx_manager.rs

```diff
@@ -634,9 +610,34 @@ impl EthTxManager {
             // sending new operations.
             let _ = self
                 .send_eth_tx(storage, &tx, time_in_mempool, l1_block_numbers.latest)
-                .await;
+                .await?;
         }
+        Ok(())
+    }
 
-        Ok(l1_block_numbers.latest)
+    #[tracing::instrument(skip(self, storage))]
+    async fn loop_iteration(
+        &mut self,
+        storage: &mut Connection<'_, Core>,
+        l1_block_numbers: L1BlockNumbers,
+    ) {
+        tracing::info!("Loop iteration at block {}", l1_block_numbers.latest);
+        for operator_type in [OperatorType::NonBlob, OperatorType::Blob] {
```

Comment:
is it possible that due to this change we'll mark a `prove` or an `execute` as complete before we do the same for `commit` of the same batch (issue similar to the one that brought ENs down some days ago?)

Reply:
Fortunately we don't have to worry about it in eth-sender, the aggregator only creates PublishProof transaction if corresponding Commit transaction is confirmed:
https://github.com/matter-labs/zksync-era/blob/main/core/node/eth_sender/src/aggregator.rs#L312
So we only have to make sure that transactions for a single operator are confirmed in-order

Reply:
It's good idea to add comment about that!

Reply:
Added a comment